logger.go

```diff
@@ -250,6 +250,21 @@ func (log *Logger) Core() zapcore.Core {
 	return log.core
 }
 
+func (log *Logger) SetCore(core zapcore.Core) {
+	log.core = core
+}
+
+func (log *Logger) Clone() *Logger {
```

Comment:
this can be achieved via `logger.With()` but generally isn't needed, as loggers are not mutable.